section_faiss_vector_index.go

```diff
@@ -76,6 +79,10 @@ func remapDocIDs(oldIDs *roaring.Bitmap, newIDs []uint64) *roaring.Bitmap {
 	return newBitmap
 }
 
+// constants for the purpose of persisting the index type as an int
+const recallIndexInt = 0
+const latencyIndexInt = 1
```

Comment:
This is very brittle, declaring different constants here. You should at least get these numbers for recall or latency by setting up values for the map you've declared here -  https://github.com/blevesearch/bleve_index_api/pull/46 .